node/src/blockchain/blockchain_interface/test_utils.rs

```diff
@@ -1,70 +1,112 @@
-    fn get_transaction_fee_balance(&self, address: &Wallet) -> ResultForBalance {
-        self.get_transaction_fee_balance_params
-            .lock()
-            .unwrap()
-            .push(address.clone());
-        self.get_transaction_fee_balance_results
-            .borrow_mut()
-            .remove(0)
+    fn get_transaction_fee_balance(
+        &self,
+        _address: Address,
+    ) -> Box<dyn Future<Item = U256, Error = BlockchainError>> {
+        unimplemented!("not needed so far")
     }
 
-    fn get_service_fee_balance(&self, address: &Wallet) -> ResultForBalance {
-        self.get_masq_balance_params
-            .lock()
-            .unwrap()
-            .push(address.clone());
-        self.get_masq_balance_results.borrow_mut().remove(0)
+    fn get_service_fee_balance(
+        &self,
+        _address: Address,
+    ) -> Box<dyn Future<Item = U256, Error = BlockchainError>> {
+        unimplemented!("not needed so far")
     }
 
-    fn get_block_number(&self) -> LatestBlockNumber {
-        self.get_block_number_results.borrow_mut().remove(0)
+    fn get_gas_price(&self) -> Box<dyn Future<Item = U256, Error = BlockchainError>> {
+        unimplemented!("not needed so far")
     }
 
-    fn get_transaction_id(&self, address: &Wallet) -> ResultForNonce {
-        self.get_transaction_id_params
-            .lock()
-            .unwrap()
-            .push(address.clone());
-        self.get_transaction_id_results.borrow_mut().remove(0)
+    fn get_block_number(&self) -> Box<dyn Future<Item = U64, Error = BlockchainError>> {
+        unimplemented!("not needed so far")
+    }
+
+    fn get_transaction_id(
+        &self,
+        _address: Address,
+    ) -> Box<dyn Future<Item = U256, Error = BlockchainError>> {
+        unimplemented!("not needed so far")
+    }
+
+    fn get_transaction_receipt(
+        &self,
+        _hash: H256,
+    ) -> Box<dyn Future<Item = Option<TransactionReceipt>, Error = BlockchainError>> {
+        unimplemented!("not needed so far")
+    }
+
+    fn get_transaction_receipt_batch(
+        &self,
+        _hash_vec: Vec<H256>,
+    ) -> Box<dyn Future<Item = Vec<TransactionReceiptResult>, Error = BlockchainError>> {
+        unimplemented!("not needed so far")
+    }
+
+    fn get_contract(&self) -> Contract<Http> {
+        unimplemented!("not needed so far")
+    }
+
+    fn get_transaction_logs(
+        &self,
+        _filter: Filter,
+    ) -> Box<dyn Future<Item = Vec<Log>, Error = BlockchainError>> {
+        unimplemented!("not needed so far")
+    }
+
+    fn submit_payables_in_batch(
+        &self,
+        _logger: Logger,
+        _chain: Chain,
+        _consuming_wallet: Wallet,
+        _fingerprints_recipient: Recipient<PendingPayableFingerprintSeeds>,
+        _affordable_accounts: Vec<PayableAccount>,
+    ) -> Box<dyn Future<Item = Vec<ProcessedPayableFallible>, Error = PayableTransactionError>>
+    {
+        unimplemented!("not needed so far")
```

Comment:
Didn't you guys had it pop up in your head that it is quite surprising why most of these methods now need to be filled with an unimplemented?

I understand you didn't want to dig too much into it but let's give it a shot. It's definitely tightly coupled with the fact you decided for quite a courageous plan of employing MockBlockchainClientServer in a big scale, in many and many tests. The result of your effort was that this mock declared above completely lost a purpose in our codebase. It's because your way works like a one-to-one substitution.

I don't feel, though, I'm ready to give you clear advice on how to react now. 

But as you can guess out, something in my head is saying, just don't be a coward and tell them to delete this whole bunch of code. (reasoning: It's not needed now, as outlived, and if it doesn't exist anymore, people will genuinely incline to your solution guys - which I'm okay with, by the way - altogether leading into some desirable consistency).

Now, I'm handing it to our architect.

Alert, we need to talk to Dan Wiebe.